src/js/utils/log.js

```diff
@@ -93,6 +99,76 @@ log = function(...args) {
 log.history = [];
 
 /**
+ * Enumeration of available logging levels, where the keys are the level names
+ * and the values are `|`-separated strings containing logging methods allowed
+ * in that logging level.
+ *
+ * This allows users to define their own logging levels if desired. Levels are
+ * used to create a regular expression matching the function name being called.
+ *
+ * Levels provided by video.js are:
+ *
+ * - `off`: Matches no calls. Any value that can be cast to `false` will have
+ *   this effect. The most restrictive.
+ * - `debug` (default): Matches only Video.js-provided functions (`log`,
+ *   `log.warn`, and `log.error`).
+ * - `warn`: Matches `log.warn` and `log.error` calls.
+ * - `error`: Matches only `log.error` calls.
+ *
+ * @type {Object}
+ */
+log.levels = {};
+
+Object.defineProperties(log.levels, {
```

Comment:
Fair point. Well, we can also simply have it use normal properties for browsers that don't support `Object.defineProperties`.